packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/maps_object.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart' show objectRuntimeType;
+import 'package:meta/meta.dart' show immutable;
+
+/// Uniquely identifies object an among [GoogleMap] collections of a specific
+/// type.
+///
+/// This does not have to be globally unique, only unique among the collection.
+@immutable
+class MapsObjectId<T> {
+  /// Creates an immutable object representing a [T] among [GoogleMap] Ts.
+  ///
+  /// An [AssertionError] will be thrown if [value] is null.
+  MapsObjectId(this.value) : assert(value != null);
+
+  /// The value of the id.
+  final String value;
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) return true;
+    if (other.runtimeType != runtimeType) return false;
+    final MapsObjectId<T> typedOther = other as MapsObjectId<T>;
+    return value == typedOther.value;
+  }
+
+  @override
+  int get hashCode => value.hashCode;
+
+  @override
+  String toString() {
+    return '${objectRuntimeType(this, 'MapsObjectId')}($value)';
```

Comment:
This is merely cosmetic, but in web the `objectRuntimeType` often gets minified to the point of not being very usable (see the example LatLng demo).

Can this class have a `displayName`  (similar to the `objectName` of MapsObjectUpdate)?

This is merely cosmetic, and the use of toSting should be limited, so "no" is a perfectly valid answer :P

Reply:
This is ~behavior preserving; since a change to the stringification could easily be made separately from this PR I'd rather not include it here.